app/src/main/java/chat/rocket/android/settings/ui/SettingsFragment.kt

```diff
@@ -163,29 +165,49 @@ class SettingsFragment : Fragment(), SettingsView, AppLanguageView {
 
     private fun changeLanguage() {
         context?.let {
+            val locales = ArrayList<Pair<String, String>>()
+            with(locales) {
+                add(Pair("en", ""))
```

Comment:
Why to create a Pair if the country for some languages don't exist? It is useless in that case.

Reply:
I used Pair for all of them since I just reference to it using `locale.first` and `locale.second` at the bottom. I can make it be an `arrayListOf<Any>()` and add String together with Pair but then it will make the implementation longer at the bottom, since I will need to check for the instance type if it is a Pair or just a String. Do you recommend using that approach I just said, or do you have any other suggestion? I am surely open to any efficient solution.

Reply:
Why creating a Pair if the `second` is useless? Actually there is no need to create a Pair if we know that some languages don't have a country code. You are just wasting it and making everyone who works with that code creating an empty string in that cases

Reply:
I have fixed it. Please review the updated code. I added String for locales with language only, and added Pair for locales with language and country. I also changed the way how the current selected locale is read. Instead of reading it from the SharedPreferences, I get it from the current applied configuration. This is useful if language and country are unspecified; that is if the user never set a language from the settings before.